### PR TITLE
[SHELL32] Enable custom CD/DVD icons

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose
This PR will properly show the custom icon for the optical disk with file "autorun.inf".
JIRA issue: [CORE-14766](https://jira.reactos.org/browse/CORE-14766)

### CODE REVIEW

**Comment by katahiromz** [comment]
AFTER:
drive-after

File: dll/win32/shell32/folders/CDrivesFolder.cpp [hunk 1]
```diff
@@ -297,6 +297,58 @@ HRESULT CDrivesContextMenu_CreateInstance(PCIDLIST_ABSOLUTE pidlFolder,
...
+{
+    WCHAR wszPath[MAX_PATH];
+    WCHAR wszAutoRunInfPath[MAX_PATH];
+    WCHAR wszValue[MAX_PATH], wszTemp[MAX_PATH];
+    static const WCHAR wszAutoRunInf[] = { 'a','u','t','o','r','u','n','.','i','n','f',0 };
+    static const WCHAR wszAutoRun[] = { 'a','u','t','o','r','u','n',0 };
+
+    // get path
+    if (!ILGetDisplayNameExW(psf, pidl, wszPath, 0))
+        goto Quit;
```
Comment (learn-more) [inline dll/win32/shell32/folders/CDrivesFolder.cpp]: There is not really something to clean up, so `return E_FAIL;` would be preferred here (imo)

File: dll/win32/shell32/folders/CDrivesFolder.cpp [hunk 2]
```diff
@@ -297,6 +297,58 @@ HRESULT CDrivesContextMenu_CreateInstance(PCIDLIST_ABSOLUTE pidlFolder,
     return CDefFolderMenu_Create2(pidlFolder, hwnd, cidl, apidl, psf, DrivesContextMenuCallback, cKeys, hKeys, ppcm);
 }
 
+static BOOL
+getAutoRunInfo(LPCWSTR Entry, LPWSTR pszValue, DWORD cchValueLen, LPCWSTR InfFile)
+{
+    static const WCHAR s_icon[] = { 'i', 'c', 'o', 'n', 0 };
```
Comment (learn-more) [inline dll/win32/shell32/folders/CDrivesFolder.cpp]: This code is not wine-synced so you can just use `L"icon"`

File: dll/win32/shell32/folders/CDrivesFolder.cpp [hunk 3]
```diff
@@ -297,6 +297,57 @@ HRESULT CDrivesContextMenu_CreateInstance(PCIDLIST_ABSOLUTE pidlFolder,
     return CDefFolderMenu_Create2(pidlFolder, hwnd, cidl, apidl, psf, DrivesContextMenuCallback, cKeys, hKeys, ppcm);
 }
 
+static BOOL
+getAutoRunInfo(LPCWSTR Entry, LPWSTR pszValue, DWORD cchValueLen, LPCWSTR InfFile)
+{
+    static const WCHAR s_icon[] = L"icon";
```
Comment (HBelusca) [inline dll/win32/shell32/folders/CDrivesFolder.cpp]: If you don't use the Wine-style of declaring unicode strings (as you do below in getIconLocationForDrive()), then you can as well put directly the string in the parameters of the GetPrivateProfileStringW() call.

**Comment by katahiromz** [comment]
Done.

**Pull request merged**